pkg/tests/apis/provisioning/connection_test.go

```diff
@@ -0,0 +1,112 @@
+package provisioning
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/grafana/grafana/pkg/util/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func TestIntegrationProvisioning_CreatingAndGettingConnections(t *testing.T) {
```

Comment:
I'm loving this integration test-first mindset :) Let's check CRUD operations already and other cases.